nixos/modules/services/networking/shairport-sync.nix

```diff
@@ -66,6 +74,9 @@ in
         extraGroups = [ "audio" ] ++ optional config.hardware.pulseaudio.enable "pulse";
       };
 
+    networking.firewall.allowedTCPPorts = if cfg.openFirewall then [ 5000 ] else [];
```

Comment:
See `mkIf`, `optional`, or `optionals` from `lib`.

Reply:
Great idea!  I've changed this to use `mkIf`.